### PR TITLE
MINIFICPP-2100 Add RockyLinux Dockerfile

### PR DESCRIPTION
**Pull request by lordgamez**
https://issues.apache.org/jira/browse/MINIFICPP-2100

--------------------
Thank you for submitting a contribution to Apache NiFi - MiNiFi C++.

In order to streamline the review of the contribution we ask you
to ensure the following steps have been taken:

### For all changes:
- [ ] Is there a JIRA ticket associated with this PR? Is it referenced
     in the commit message?

- [ ] Does your PR title start with MINIFICPP-XXXX where XXXX is the JIRA number you are trying to resolve? Pay particular attention to the hyphen "-" character.

- [ ] Has your PR been rebased against the latest commit within the target branch (typically main)?

- [ ] Is your initial contribution a single, squashed commit?

### For code changes:
- [ ] If adding new dependencies to the code, are these dependencies licensed in a way that is compatible for inclusion under [ASF 2.0](http://www.apache.org/legal/resolved.html#category-a)?
- [ ] If applicable, have you updated the LICENSE file?
- [ ] If applicable, have you updated the NOTICE file?

### For documentation related changes:
- [ ] Have you ensured that format looks appropriate for the output in which it is rendered?

### Note:
Please ensure that once the PR is submitted, you check GitHub Actions CI results for build issues and submit an update to your PR as soon as possible.

### CODE REVIEW

File: docker/rockylinux/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+# gpsd-devel and ccache are in EPEL
+RUN yum -y install epel-release && yum -y install sudo git which make libarchive ccache ca-certificates gcc-toolset-11-toolchain && \
+    if echo "$MINIFI_OPTIONS" | grep -q "ENABLE_GPS=ON"; then yum -y install gpsd-devel; fi && \
+    if echo "$MINIFI_OPTIONS" | grep -q "ENABLE_JNI=ON"; then yum -y install java-1.8.0-openjdk maven; fi && \
+    if echo "$MINIFI_OPTIONS" | grep -q "ENABLE_PCAP=ON"; then yum -y install libpcap-devel; fi && \
+    if echo "$MINIFI_OPTIONS" | grep -q "ENABLE_USB_CAMERA=ON"; then yum -y install libpng-devel libusbx-devel; fi && \
+    if echo "$MINIFI_OPTIONS" | grep -q "ENABLE_OPENCV=ON" || echo "$MINIFI_OPTIONS" | grep -q "ENABLE_BUSTACHE=ON"; then yum -y install boost-devel; fi && \
+    if echo "$MINIFI_OPTIONS" | grep -q "ENABLE_LUA_SCRIPTING=ON"; then yum -y install lua-devel; fi && \
+    if echo "$MINIFI_OPTIONS" | grep -q "ENABLE_PYTHON_SCRIPTING=ON"; then yum -y install python36-devel; fi && \
+    if echo "$MINIFI_OPTIONS" | grep -q "ENABLE_SFTP=ON" && [ "${DOCKER_SKIP_TESTS}" == "OFF" ]; then yum -y install java-1.8.0-openjdk maven; fi
```
[the remark this answers is not in the document]
Comment (martinzink) [reply docker/rockylinux/Dockerfile:49]: We could also install lua-devel without enableing crb with something like this
`dnf --enablerepo=powertools install lua-devel`

[the remark this answers is not in the document]
Comment (lordgamez) [reply docker/rockylinux/Dockerfile]: Updated in b04a27f9a52eb3f0cdd0c637aa815b5e0183d93f

File: docker/rockylinux/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+# Setup minificpp user
+RUN groupadd -g ${GID} ${USER} && useradd -g ${GID} ${USER} && \
+    chown -R ${USER}:${USER} ${MINIFI_BASE_DIR}
+
+USER ${USER}
+
+# Perform the build
+RUN cd $MINIFI_BASE_DIR && \
+    cd build && \
+    source /opt/rh/gcc-toolset-11/enable && \
```
[the remark this answers is not in the document]
Comment (lordgamez) [reply docker/rockylinux/Dockerfile]: Updated in b04a27f9a52eb3f0cdd0c637aa815b5e0183d93f

**Review by martinzink — APPROVED** [review]

File: docker/rockylinux/Dockerfile [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+
+# Install the system dependencies needed for a build
+# gpsd-devel and ccache are in EPEL
+RUN dnf -y install epel-release && dnf -y install sudo git which make libarchive ccache ca-certificates cmake && \
+    if echo "$MINIFI_OPTIONS" | grep -q "ENABLE_GPS=ON"; then dnf -y install gpsd-devel; fi && \
+    if echo "$MINIFI_OPTIONS" | grep -q "ENABLE_JNI=ON"; then dnf -y install java-1.8.0-openjdk maven; fi && \
+    if echo "$MINIFI_OPTIONS" | grep -q "ENABLE_PCAP=ON"; then dnf -y install libpcap-devel; fi && \
+    if echo "$MINIFI_OPTIONS" | grep -q "ENABLE_USB_CAMERA=ON"; then dnf -y install libpng-devel libusbx-devel; fi && \
+    if echo "$MINIFI_OPTIONS" | grep -q "ENABLE_OPENCV=ON" || echo "$MINIFI_OPTIONS" | grep -q "ENABLE_BUSTACHE=ON"; then dnf -y install boost-devel; fi && \
+    if echo "$MINIFI_OPTIONS" | grep -q "ENABLE_LUA_SCRIPTING=ON"; then dnf --enablerepo=powertools install lua-devel; fi && \
```
Comment (martinzink) [inline docker/rockylinux/Dockerfile:47]: sry I might have left out the noprompt flag from here
```suggestion
    if echo "$MINIFI_OPTIONS" | grep -q "ENABLE_LUA_SCRIPTING=ON"; then dnf -y --enablerepo=powertools install lua-devel; fi && \
```

**Review by martinzink — CHANGES_REQUESTED** [review]

File: docker/rockylinux/Dockerfile (hunk 3 above)
Comment (lordgamez) [reply docker/rockylinux/Dockerfile]: Good catch, updated in 1ca67cbec06fca4ed5fe229740cf0f9d3c7adb11

**Review by martinzink — APPROVED** [review]

**Review by fgerlits — APPROVED** [review]

**Pull request closed without merging**